moesifwsgi/workers.py

```diff
@@ -173,3 +175,45 @@ def stop(self):
         # Clear workers
         self.batcher = None
         self.workers = []
+
+
+class ConfigJobScheduler:
+
+    def __init__(self, debug, config):
+        self.DEBUG = debug
+        self.scheduler = None
+        self.config = config
+
+    def exit_config_job(self):
+        try:
+            # Shut down the scheduler
+            self.scheduler.remove_job('moesif_config_job')
+            self.scheduler.shutdown()
+        except Exception as ex:
+            if self.DEBUG:
+                print("Error during shut down of the config scheduler")
+                print(str(ex))
+
+    def schedule_background_job(self):
+        try:
+            if not self.scheduler:
+                self.scheduler = BackgroundScheduler(daemon=True)
+            if not self.scheduler.get_jobs():
```

Comment:
@keyur9 do we need to check if get_jobs() is `truthy` but job status is cancelled or terminated? In that case we need to clean up and reschedule?

Reply:
we can take care of it in next release.

Reply:
get_jobs return the list of pending and scheduled jobs, so we're checking if no job is pending or scheduled then only we add the config job. So there won't be any need to check the status